### PR TITLE
Add ghcjs to CI and fix build issues

### PR DESCRIPTION
**Pull request by hamishmack**
[Hydra jobset PR #106](https://hydra.iohk.io/jobset/Cardano/cardano-addresses-pr-106#tabs-jobs)

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -14,6 +14,13 @@ source-repository-package
     tag: 2547ad1e80aeabca2899951601079408becbc92c
     --sha256: 1p2kg2w02q5w1cvqzhfhqmxviy4xrzada3mmb096j2n6hfr20kri
 
+source-repository-package
+    type: git
+    location: https://github.com/hamishmack/foundation
```
Comment (rvl) [inline cabal.project:19]: Because this build fix is so small, could we do it as a patch instead of forking on github?

File: nix/sources.json [hunk 2]
```diff
@@ -1,14 +1,14 @@
 {
     "haskell.nix": {
-        "branch": "master",
+        "branch": "hkm/emcc",
```
Comment (rvl) [inline nix/sources.json:3]: Fine to use a branch for now - I assume there will be a PR some time?

Does this mean it actually uses emcc to compile the cbits?

**Review by rvl — APPROVED** [review]
Awesome, thanks!

File: nix/sources.json (hunk 2 above)
Comment (hamishmack) [reply nix/sources.json:3]: I'm still not sure I have it right.  Will check with @luite.  Once it is we can merge it and set this back to master.

File: cabal.project (hunk 1 above)
Comment (hamishmack) [reply cabal.project:19]: I looked into this and it won't work as a patch.  We need `cabal configure` to see the patched version and haskell.nix can only patch the sources selected after `cabal configure` runs.

Comment (rvl) [reply cabal.project:19]: OK - fair enough!

File: nix/sources.json (hunk 2 above)
Comment (rvl) [reply nix/sources.json:3]: OK, so #108.

**Pull request merged**